### ISSUE TITLE
(aws_rds): Native support for Blue Green deployments of RDS via CDK

### ISSUE THREAD

**Issue by mrlikl** [aws/aws-cdk#23872]
### Describe the feature

Raising this ticket to check if native support for a blue/green deployment of RDS will be added to CDK 

### Use Case

Implementation of this feature that is available in RDS
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/blue-green-deployments-creating.html

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

NA

### Environment details (OS name and version, etc.)

NA

**Comment by pahud** [comment]
This is interesting. What CDK experience would you expect for that? Can you elaborate more about it?

**Comment by mrlikl** [comment]
LIke a custom resource leveraging the API call CreateBlueGreenDeployment

**Comment by bdekonin** [comment]
Are there any plans to implement this in the future?

**Comment by ariciputi** [comment]
I can see this issue hasn't received much attention, however as already pointed out above having a CDK resource implementing the `CreateBlueGreenDeployment`, `DescribeBlueGreenDeployments`, `SwitchoverBlueGreenDeployment`, and the `DeleteBlueGreenDeployment` API calls might be useful to orchestrate the entire Blue/Green lifecycle from CDK.

It would be nice to know if there is any chance for these APIs to be supported in the foreseeable future.

Cheers.

**Comment by kiliancm94** [comment]
I am also interested in this topic. Are there plans to work on it?

**Comment by ml-deb** [comment]
Also interested. 

I found this article, maybe it can help while we wait for the feature: https://ruempler.eu/2023/10/08/zero-downtime-upgrade-aws-aurora-mysql5-7-8-0-with-blue-green/

**Comment by dmxkov** [comment]
If we can have this feature implemented into CDK then nightmares with updating even minor version will go away.
At this point we're using manual process and CLI but would be nice to have it fully automated within CDK.

**Comment by mrlikl** [comment]
I was testing the behaviour from CloudFormation, when B/G is enabled manually out of the stack there are no drifts introduced. However, during deletion CloudFormation is not able to clean up all the resources. There will be one instance left in the account and if there are dependent resources like DBParameterGroup  created via the stack, that will also fail deletion due to `resource in use`.

**Comment by blimmer** [comment]
See also https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1483

**Comment by Koketjo001** [comment]
Hello all,  are there any plans to have this implemented ? Do we have any workarounds to upgrade a major RDS version without a downtime?

**Comment by Ram-codee** [comment]
+1 to this feature request. 
We have created around 20+ env/system through cdk and it's very challenging(mostly impossible to upgrade using blue/green method). Also we can't tolerate the downtime to follow the other upgrde methods.

Is there any traction on this feature /ETA?

**Comment by mrgrain** [comment]
> https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1483

CDK is currently blocked by this CloudFormation roadmap item. Please let your account managers know if you would like this CloudFormation feature to be implemented.

**Comment by itsmaryann** [comment]
+1 to this feature request.
